File: backend/schemas/v1/external_link.py
from pydantic import BaseModel
from typing import Optional
from schemas.v1.shared.external_site import ExternalSite

class ExternalLink(BaseModel):
    id: int
    anime_id: int
    site_id: int
    url: str
    external_site: ExternalSite  # Asegúrate de que este campo esté presente y correcto
    
    class Config:
        from_attributes = True

File: frontend/src/types/shared/ExternalLink.ts
export interface ExternalLink {
  id: number
  anime_id: number
  site_id: number
  url: string
  external_site: ExternalSite
}

export interface ExternalSite {
  id: number
  name: string
  color?: string
  icon?: string
  type: string
  language?: string
  isDisabled: boolean
}
